### PR TITLE
# 167916033 Ch add h cases 

### PR DESCRIPTION
**Pull request by nziokaivy**
#### What does this PR do?
Add H cases

#### Description of Task to be completed?
- Ensure a user cannot create a booking twice 
- Ensure a user cannot create a trip twice
- Combine all bookings by user and admin routes into one get bookings route

#### How should this be manually tested?
CLone this repository and run `npm install && run test`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#167916033

#### Screenshots (if appropriate)
Screenshot from 2019-08-15 19-07-37
Screenshot from 2019-08-15 19-12-53
Screenshot from 2019-08-15 19-11-13
Screenshot from 2019-08-15 19-15-58

### CODE REVIEW

**Comment by oesukam** [comment]
@nziokaivy Travis is failing and your PR's title doesn't follow the required format

File: server/api/controllers/bookingController.js [hunk 1]
```diff
@@ -6,47 +6,79 @@ class BookingController {
...
+		const decodedToken = jwt.verify(token, process.env.JWT_KEY);
+		req.body.data = decodedToken;
+		const { is_admin } = req.body.data;
+		if (is_admin) {
+			const allBookings = Booking.getAllBookings();
+			console.log(await allBookings);
+			
+			if (await allBookings) {
+				return res.status(200).json({
+					status: '200',
```
Comment (oesukam) [inline server/api/controllers/bookingController.js]: Status is an integer

File: server/api/controllers/bookingController.js [hunk 2]
```diff
@@ -6,47 +6,79 @@ class BookingController {
...
+					message: 'success',
+					data: await allBookings,
+				});
+			}
+			return res.status(404).json({
+				status: '404',
```
Comment (oesukam) [inline server/api/controllers/bookingController.js]: Status is an integer

File: server/api/controllers/tripController.js [hunk 3]
```diff
@@ -2,64 +2,64 @@ import Trip from '../models/trip';
 
 class TripController {
 	static async createNewTrip(req, res) {
-		const { body } = req;
-		const newTrip = Trip.createNewTrip({ ...body });
-		// const checkTrip = Trip.checkRepeatTrip(body.bus_license_number, body.trip_date);
+		const {
+			body
+		} = req;
+		console.log(body);
```
Comment (oesukam) [inline server/api/controllers/tripController.js]: Remove the log

File: server/api/controllers/bookingController.js [hunk 4]
```diff
@@ -6,47 +6,79 @@ class BookingController {
...
 	static async deleteBooking(req, res) {
-		const token = req.headers.authorization.split(' ')[1];
-		const decodedToken = jwt.verify(token, process.env.JWT_KEY);
-		req.body.data = decodedToken;
 		const bookingId = parseInt(req.params.id);
 		const findBooking = Booking.deleteBooking(bookingId);
 		if (await findBooking) {
-			return res.status(200).json({ status: '204', data: { message: 'Booking Deleted Successfully!' } });
+			return res.status(200).json({
+				status: '204',
```
Comment (oesukam) [inline server/api/controllers/bookingController.js]: Status's type not correct

File: server/api/controllers/tripController.js [hunk 5]
```diff
@@ -2,64 +2,64 @@ import Trip from '../models/trip';
...
 		});
 	}
 
 	static async getSpecificTrip(req, res) {
 		const id = parseInt(req.params.id);
 		const specificTrip = Trip.getSpecificTrip(id);
+		console.log(await specificTrip);
 		if (await specificTrip) {
 			return res.status(200).json({
 				status: 'success',
```
Comment (oesukam) [inline server/api/controllers/tripController.js]: Status should be an integer

File: server/api/controllers/tripController.js [hunk 6]
```diff
@@ -2,64 +2,64 @@ import Trip from '../models/trip';
...
 			});
 		}
-		if (!specificTrip) {
-			return res.status(404).json({
-				status: 'error',
-				error: 'Not found',
-			});
-		}
+		return res.status(404).json({
+			status: 'error',
```
Comment (oesukam) [inline server/api/controllers/tripController.js:53]: Should be an integer

File: server/api/db/Db.js [hunk 7]
```diff
@@ -7,7 +7,7 @@ dotenv.config();
 const pool = {
 	connectionString: config.db,
 };
-// eslint-disable-next-line no-console
+
 console.log(config.db);
```
Comment (oesukam) [inline server/api/db/Db.js]: Remove log

File: server/test/trips.test.js [hunk 8]
```diff
@@ -169,13 +147,64 @@ describe('Trip Tests', () => {
...
+
+	// TEST FOR CREATE A NEW TRIP 
+	it('POST/api/v1/trips Should create a new trip', (done) => {
+		const trip = {
+			seating_capacity: '23',
+			bus_license_number: 'KZE 432',
+			origin: 'Nairobijhjjghghghbhjbj',
+			destination: 'KIgalijhjjjhug',
+			trip_date: '21/09/2019',
+			fare: '4000',
```
Comment (oesukam) [inline server/test/trips.test.js:185]: Fare isn't a string

File: server/test/trips.test.js [hunk 9]
```diff
@@ -169,13 +147,64 @@ describe('Trip Tests', () => {
...
+	});
+
 	// TEST FOR GET ALL TRIPS
 	it('GET/api/v1/trips Should fetch all trips', (done) => {
 		chai
 			.request(app)
 			.get('/api/v1/trips')
 			.set('authorization', `Bearer ${userToken}`)
 			.end((err, res) => {
+				console.log(res.body);
```
Comment (oesukam) [inline server/test/trips.test.js]: Remove the log

**Review by oesukam — CHANGES_REQUESTED** [review]

File: server/test/trips.test.js (hunk 9 above)
Comment (nziokaivy) [reply server/test/trips.test.js]: Removed.

**Comment by nziokaivy** [comment]
> @nziokaivy Travis is failing and your PR's title doesn't follow the required format

Solved this issue.

> @nziokaivy Travis is failing and your PR's title doesn't follow the required format

@oesukam I have now worked on it

File: server/api/controllers/bookingController.js (hunk 1 above)
Comment (nziokaivy) [reply server/api/controllers/bookingController.js]: Changed it

File: server/api/controllers/bookingController.js (hunk 2 above)
Comment (nziokaivy) [reply server/api/controllers/bookingController.js]: Changed it

File: server/api/controllers/tripController.js (hunk 3 above)
Comment (nziokaivy) [reply server/api/controllers/tripController.js]: Removed

File: server/api/controllers/bookingController.js (hunk 4 above)
Comment (nziokaivy) [reply server/api/controllers/bookingController.js]: Changed it

File: server/api/controllers/tripController.js (hunk 5 above)
Comment (nziokaivy) [reply server/api/controllers/tripController.js]: Edited it.

File: server/api/controllers/tripController.js (hunk 6 above)
Comment (nziokaivy) [reply server/api/controllers/tripController.js:53]: Edited it.

File: server/api/db/Db.js (hunk 7 above)
Comment (nziokaivy) [reply server/api/db/Db.js]: Removed it.

File: server/test/trips.test.js (hunk 8 above)
Comment (nziokaivy) [reply server/test/trips.test.js:185]: My validation for the fare I used regex which takes in strings.

**Pull request merged**